docs/testplan/GCU-Dynamic-ACL-testplan.md

```diff
@@ -0,0 +1,464 @@
+
+#### Test Objective
+
+Verify that GCU is capable of adding a large amount of ACL rules and that they are still followed.
+
+#### Testing Steps
+
+- Use GCU to create 150 forwarding rules on ACL Table
+
+- Use GCU to create drop rules for each server facing port on ACL Table
+
+- Verify that all operations were successful
+
+- Verify that rules created are properly shown when using "show acl rule"
+
+- Verify that packets with IPs matching a destination IP for one of the forwarding rules are forwarded
+
+- Verify that packets not matching a forwarding IP are dropped
+
+### Test Case # 8 - Replace the IP Address of a non-existent ACL Rule
+
+#### Test Objective
+
+Verify that attempting to replace the address of a rule that does not exist properly results in an error and does not affect configDB
+
+#### Testing Steps
+
+- Create 2 new forwarding rules on ACL Table
+
+- Replace the IP addresses in non-existent forwarding rules
+
+- Verify that the replace action failed
+
+### Test Case # 9 - Remove non-existent ACL Table
+
+#### Test Objective
+
+Verify that attempting to remove an ACL Table that does not exist properly results in an error and does not affect configDB
+
+#### Testing Steps
+
+- Attempt to remove a table that does not exist
+
+- Verify that this removal fails
+
+## JSON Patch Files and Expected Results
+
+This section contains explicit details on the contents of each JSON Patch file used within the test, as well as the exact way that these operations are  checked for success
+
+### Create a new ACL table type
+**JSON Patch:**
+
+	[
+        {
+            "op": "add",
+            "path": "/ACL_TABLE_TYPE",
+            "value": {
+                "DYNAMIC_ACL_TABLE_TYPE" : {
+                "MATCHES": ["DST_IP","DST_IPV6","IN_PORTS"],
+                "ACTIONS": ["PACKET_ACTION","COUNTER"],
+                "BIND_POINTS": ["PORT"]
+                }
+            }
+        }
+    ]
+**Expected Result**
+- Operation Success
+
+**Additional checks**
+- None
+
+### Create an ACL Table
+
+**Json Patch**:
+
+    [
+        {
+            "op": "add",
+            "path": "/ACL_TABLE/DYNAMIC_ACL_TABLE",
+            "value": {
+                "policy_desc": "DYNAMIC_ACL_TABLE",
+                "type": "DYNAMIC_ACL_TABLE_TYPE",
+                "stage": "INGRESS",
+                "ports": {vlan port members from minigraph}
+            }
+        }
+    ]
+
+**Expected result**
+- Operation Success
+
+
+**Additional checks**
+- Check that results of the command “show acl table” match this expected output:
+
+Name  | Type | Binding | Description | Stage | Status
+------------- | ------------- | ---------- | ----------| --------- | ---------
+DYNAMIC_ACL_TABLE | DYNAMIC_ACL_TABLE_TYPE | {vlan port 1} | DYNAMIC_ACL_TABLE_TYPE | ingress | Active
+  | | {vlan port 2} | |
+  | | {vlan port 3}...
+
+### Create Forwarding Rules
+
+**Json Patch:**
+
+    [
+        {
+            "op": "add",
+            "path": "/ACL_RULE",
```

Comment:
>ACL_RULE [](http://example.com/codeflow?start=22&length=8)

There may be large amount existing rules inside ACL_RULE table. if you "add" at this "path", by RFC, you actually delete all existing rules and add "DYNAMIC_ACL_TABLE" rules. Is this your intention? How do you revert back after finish your test?

Reply:
Before tests start, a fixture creates a checkpoint of the initial state, and then rolls back to it after tests are complete.

The patch is formatted this way because forward rules are always the first rules added in any test that they are involved in.  Without formatting it this way, 'config apply-patch' errors out because the table location does not exist